contract_show_recurring_invoice/__init__.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from . import models
+
+
+def copy_recurring_invoice(cr, registry):
+    """Copy recurring invoice on contract."""
+    cr.execute("UPDATE account_invoice "
```

Comment:
Use a compose UPDATE:
```
UPDATE account_invoice
SET contract_id=aaa.id
FROM account_analytic_account aaa
WHERE aaa.code = account_invoice.origin
```
More efficient and you avoid undetermined cases when there's no match.